### DISCUSSION TITLE
Should we do database split from zebra state and zebra-db as pat of the zebra-scan MVP ?

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#7985] [Q&A] [answered]
We want to store persistent data (keys and results) as part of the MVP but some of us are unsure if the best way to go first is to do https://github.com/ZcashFoundation/zebra/issues/7926 and https://github.com/ZcashFoundation/zebra/issues/7937 or to just create a database duplicating code as needed inside zebra-scan leaving zebra-state untouched and no new crate creation.

**Comment by teor2345** [comment]
This is already being discussed here?

https://github.com/ZcashFoundation/zebra/discussions/7975#discussioncomment-7636414

**Comment by oxarbitrage** [comment]
That discussion has too much information and was not intended for the question asked here , can you make a resume?

**Comment by oxarbitrage** [comment]
As far as i know you were already working on splitting the zebra-state code to a new crate, How far are we from that ? If we are close already we might want to do that first.

**Reply by teor2345** [reply]
I have some draft code based on this suggestion here:
https://github.com/ZcashFoundation/zebra/discussions/7975#discussioncomment-7636554

We don't need to split the state straight away, we can give external crates access to the raw database type instead. (`DiskDb`, or maybe `ZebraDb`, I'm still working that out.)

Then after we know which parts of the database code we want to use, we can split the state later.

**Reply by mpguerra** [reply]
if we don't need to split state and it's quicker to give access to the raw database, let's do that. Do we need to create a new issue for this?

**Reply by teor2345** [reply, marked as answer]
I made a new ticket here: #8001

**Comment by oxarbitrage** [comment]
Lets close this one, thank you @teor2345
